### PR TITLE
feat: network settings

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes BX-1065
Figma link (if any):

## What changed (plus any additional context for devs)

adding network settings
users will still be able to connect to apps in rainbow supported networks, here the tx will be visible only while the tx is pending, after is confirmed it will disappear from the list
assets and txs are filtered by chains selected by the user
assets in send and swap are filtered by chains selected by the user
networks are draggable in settings, added a quick promo saying that you can drag and drop networks there
networks order should be respected all over the app

## Screen recordings / screenshots

https://www.loom.com/share/ef664f9d645a4b85a2f3532c8a986d7d

network ordering
https://www.loom.com/share/f15379ad6c2d4ba8b0238d3e3b3ff274

updated network settings with bottom text
image

### CODE REVIEW

**Review by brunobar79 — APPROVED** [review]
🔥 🔥 🔥

**Review by greg-schrammel — APPROVED** [review]
looks good

File: src/entries/popup/hooks/useInfiniteTransactionList.ts [hunk 1]
```diff
@@ -55,6 +57,20 @@ export default function ({
...
+          return {
+            ...page,
+            transactions: page.transactions.filter(
+              (transaction) => userChains[transaction.chainId],
+            ),
+          };
+        });
+        return {
+          ...data,
+          pages: selectedPages,
```
[the remark this answers is not in the document]
Comment (derHowie) [reply src/entries/popup/hooks/useInfiniteTransactionList.ts:71]: lgtm

**Review by derHowie — APPROVED** [review]
👌

**Pull request merged**